loader.js

```diff
@@ -8,9 +8,13 @@ var NodeTargetPlugin = require("webpack/lib/node/NodeTargetPlugin");
 var LibraryTemplatePlugin = require("webpack/lib/LibraryTemplatePlugin");
 var SingleEntryPlugin = require("webpack/lib/SingleEntryPlugin");
 var LimitChunkCountPlugin = require("webpack/lib/optimize/LimitChunkCountPlugin");
+
+var extractTextPluginHmrRuntime = require.resolve("./hotModuleReplacement.js");
+
 module.exports = function(source) {
 	if(this.cacheable) this.cacheable();
-	return source;
+	// Even though this gets overwritten if extract+remove are true, without it, the runtime doesn't get added to the chunk
+	return "if(module.hot){require('" + extractTextPluginHmrRuntime + "');}\n" + source;
```

Comment:
This could be a template string.

Is there any objection to using modern syntax in files that haven't yet been "converted"? I don't really see a reason to wait until a conversion; it's not like all features have to be adopted at the same time and with 100% coverage.